### DISCUSSION TITLE
HOWTO: How can I take the final LLM chain response (text) and send that as input to a Custom JS Function node?

### DISCUSSION THREAD

**Discussion by dkindlund** [FlowiseAI/Flowise#1489] [Q&A]
I'd like to ship final chatbot responses over to a custom Slack channel (for example).  What's the best/easiest way to accomplish this using like a Custom JS Function node?

**Comment by HenryHengZJ** [comment]
not do-able entirely on Flowise for now. we're working on a feature that allows sequential flow (different from chatflow), that way you can pipe the output to custom function which post to slack channel

**Reply by dkindlund** [reply]
After thinking about this more, it seems like the main requirement is just getting a Custom JS Function to execute *after* a chatflow completes.  If the Custom JS function has access to the standard `$flow` variables, then it should be (theoretically) easy for the JS function to access the full chat history to fetch the latest message through internal Flowise APIs.

For example, if I use the "Conversational Retrieval Agent" as the root node, I see that the final output dot is `AgentExecutor` like this:
<img width="477" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/85660/bb347a8e-c405-4248-98eb-a4d37382fce6">

If there were *another* output dot called `json`, then I could connect up the Custom JS Function node to that.  It does break the assumption that the "final" node is of type `LLMChain` or `Agent`, though...

**Comment by jfbloom22** [comment]
I am looking into this as well. Any updates?
